nvme-print.c

```diff
@@ -1224,6 +1224,74 @@ void show_smart_log(struct nvme_smart_log *smart, unsigned int nsid, const char
 	printf("Thermal Management T2 Total Time    : %u\n", le32_to_cpu(smart->thm_temp2_total_time));
 }
 
+void show_self_test_log(struct nvme_self_test_log *self_test, const char *devname)
+{
+	int i, temp;
+	const char *test_code_res;
+	const char *test_res[10] = {
+		"Operation completed without error",
+		"Operation was aborted by a Device Self-test command",
+		"Operation was aborted by a Controller Level Reset",
+		"Operation was aborted due to a removal of a namespace from the namespace inventory",
+		"Operation was aborted due to the processing of a Format NVM command",
+		"A fatal error or unknown test error occurred while the controller was executing the"\
+		" device self-test operation andthe operation did not complete",
+		"Operation completed with a segment that failed and the segment that failed is not known",
+		"Operation completed with one or more failed segments and the first segment that failed "\
+		"is indicated in the SegmentNumber field",
+		"Operation was aborted for unknown reason",
+		"Reserved"
+	};
+
+	printf("Smart Log for NVME device:%s\n", devname);
+	printf("Current self test operation : %#x\n", self_test->crnt_dev_selftest_oprn);
+	printf("Current Device Self-Test Completion : %u%%\n", self_test->crnt_dev_selftest_compln);
+	for (i = 0; i < NVME_SELF_TEST_REPORTS; i++) {
+		printf("\nPrinting the contents of REPORT-%d\n",(i+1));
+
+		temp = self_test->result[i].device_self_test_status & 0xf;
+		if (temp == 0xf){
+			printf("Entry not used (does not contain a test result)\n");
+			continue;
+		}
+		if(temp > 8)
+			temp = 9;
+		printf("Self test result - %#x: %s\n", temp, test_res[temp]);
```

Comment:
I understand why you're over-writing 'temp' to 9 to get the "Reserved" test result, but I'd really like to see the value that the drive actually returned here.